core/deployment/src/main/java/io/quarkus/deployment/ArthasConfig.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Configuration related to <a href="https://arthas.aliyun.com">Arthas</a>.
+ *
+ * These configuration values only apply to dev-mode.
+ */
+@ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class ArthasConfig {
+
+    /**
+     * Whether Arthas should be started when Quarkus dev-mode is launched
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean enabled;
+
+    /**
+     * The versions of Arthas to use
+     */
+    @ConfigItem(defaultValue = "3.5.5")
+    public String version;
+
+    /**
+     * The directory into which to save the arthas-boot jar if it doesn't exist
+     */
+    @ConfigItem(defaultValue = "${user.home}/.quarkus")
```

Comment:
@aloubyansky just wanted to check that you were not using this directory for other things that could interfere with this?

Reply:
Maybe use temp dir to have it automatically deleted at JVM exit ?

Reply:
The CLI uses the file `~/.quarkus/config.yaml` for adding new registries

Reply:
Right, let's not trash it. Perhaps it should have its own subdirectory at least.

Reply:
It does - it's added under `arthas`

Reply:
definitely not a fan of us downloading it into ~/.quarkus - if we really want to do this I would resolve it as dependency.

Reply:
As I mentioned in another comment - the dependency is not enough.

Reply:
Scratch that, see my comment below about the all-in-one jar.